src/v/kafka/server/handlers/produce.cc

```diff
@@ -462,6 +468,11 @@ produce_handler::handle(request_context ctx, ss::smp_service_group ssg) {
     produce_request request;
     request.decode(ctx.reader(), ctx.header().version);
 
+    if (ctx.metadata_cache().should_reject_writes()) {
+        vlog(klog.info, "Rejecting produce request: no disk space.");
```

Comment:
What do you think about logging this message with `warn` and adding some additional context coming from the request in here ? Also some time ago we chatted about rejecting requests in `connection_context` to make sure that we may reject all other relevant requests like `CreateTopics` or `CreatePartitions`

Reply:
I think this could use some rate limiting on the log message. We already log ERROR when the threshold is passed, and that log message is rate limited. We had a long discussion on blocking up in connection_context (here and on slack), and generally agreed that is the wrong layer to make produce policy decisions and responses.

Reply:
I like the idea of adding more context--wasn't sure how to access, say, client IP or other state we might want to display.